src/IB/CIBStochasticStokesSolver.cpp

```diff
@@ -0,0 +1,677 @@
+    if (d_is_initialized)
+    {
+        d_reinitializing_solver = true;
+        deallocateSolverState();
+    }
+
+    // Wrap Eulerian data into PETSc Vecs.
+    Pointer<PatchHierarchy<NDIM> > hierarchy = x.getPatchHierarchy();
+    const int coarsest_ln = x.getCoarsestLevelNumber();
+    const int finest_ln = x.getFinestLevelNumber();
+
+    for (int ln = coarsest_ln; ln <= finest_ln; ++ln)
+    {
+        Pointer<PatchLevel<NDIM> > level = hierarchy->getPatchLevel(ln);
+        if (!level->checkAllocated(d_wide_u_idx)) level->allocatePatchData(d_wide_u_idx);
+        if (!level->checkAllocated(d_wide_f_idx)) level->allocatePatchData(d_wide_f_idx);
+    }
+
+    Pointer<CellVariable<NDIM, double> > x_p_cc_var = x.getComponentVariable(1);
+    Pointer<CellVariable<NDIM, double> > b_p_cc_var = b.getComponentVariable(1);
+    const int x_p_idx = x.getComponentDescriptorIndex(1);
+    const int b_p_idx = b.getComponentDescriptorIndex(1);
+
+    d_x_wide = new SAMRAIVectorReal<NDIM, double>(x.getName() + "_wide_x", hierarchy, coarsest_ln, finest_ln);
+    d_b_wide = new SAMRAIVectorReal<NDIM, double>(b.getName() + "_wide_b", hierarchy, coarsest_ln, finest_ln);
+
+    d_x_wide->addComponent(d_wide_u_var, d_wide_u_idx, x.getControlVolumeIndex(0));
+    d_x_wide->addComponent(x_p_cc_var, x_p_idx, x.getControlVolumeIndex(1));
+    d_b_wide->addComponent(d_wide_f_var, d_wide_f_idx, b.getControlVolumeIndex(0));
+    d_b_wide->addComponent(b_p_cc_var, b_p_idx, b.getControlVolumeIndex(1));
+
+    // Wrap SAMRAI vector into PETSc Vec
+    Vec u_p = PETScSAMRAIVectorReal::createPETScVector(d_x_wide);
+    Vec g_h = PETScSAMRAIVectorReal::createPETScVector(d_b_wide);
+
+    // Get the Lagrange multiplier that maintains the rigidity constraint.
+    // NOTE: The current time corresponds to the time at which solver is initialized
+    // which maybe different from the current time of the timestep being integrated upon.
+    Vec L;
+    d_cib_strategy->getConstraintForce(&L, d_current_time);
+
+    // Create a vector of the type imposed velocity at the material/nodal points for RHS.
+    // NOTE: In the initialization stage we do not need an actual velocity vector, a reference
+    // to L should suffice to know the required structure.
+    Vec V = L;
+
+    // Get the rigid body velocities that need to be solved for.
+    // NOTE: The current time corresponds to the time at which solver is initialized
+    // which maybe different from the current time of the timestep being integrated upon.
+    Vec U;
+    d_cib_strategy->getFreeRigidVelocities(&U, d_current_time);
+
+    // Create a vector that contains net external force and torque on the body.
+    // NOTE: In the initialization stage we do not need an actual force vector, a reference
+    // to U should suffice to know the required structure.
+    Vec F = U;
+
+    // Create the composite vectors.
+    Vec mv_x, mv_b;
+    std::vector<Vec> vx(3), vb(3);
```

Comment:
We should just use `std::array<Vec, 3>` here.

Reply:
agreed

Reply:
```suggestion
    std::array<Vec,3> vx, vb;
```